actions/bump-and-release-cloud-plugin/action.yaml

```diff
@@ -0,0 +1,308 @@
+        mv dist ${{ steps.metadata.outputs.plugin-id }}
+        zip ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.plugin-id }} -r
+        md5sum ${{ steps.metadata.outputs.archive }} > ${{ steps.metadata.outputs.archive-checksum }}
+
+        # move assets to upload folder for mass upload
+        mkdir ${{ runner.temp }}/plugin-artifacts
+        mv ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.archive-checksum }} ${{ runner.temp }}/plugin-artifacts/
+
+    - name: Create latest zip
+      id: latest-zip
+      if: inputs.release-level != 'prerelease'
+      shell: bash
+      run: |
+        cd ${{ runner.temp }}/plugin-artifacts
+        cp ${{ steps.metadata.outputs.archive }} ${{ steps.metadata.outputs.archive-latest }}
+        md5sum ${{ steps.metadata.outputs.archive-latest }} > ${{ steps.metadata.outputs.archive-latest-checksum }}
+        cd ..
+
+    - name: Validate plugin
+      shell: bash
+      run: |
+        git clone https://github.com/grafana/plugin-validator
+        pushd ./plugin-validator/pkg/cmd/plugincheck2
+        go install
+        popd
+        plugincheck2 -config ./.config/plugin-validator-config.yaml ${{ runner.temp }}/plugin-artifacts/${{ steps.metadata.outputs.archive }}
+
+    # Now that we've validated our plugin we can safely commit the version bump
+    - name: Commit version bump
+      id: commit-version-bump
+      uses: grafana/github-api-commit-action@ccf9b520c5698380ad3b9619c5add427369b7ef1 # v0.2.0
+      with:
+        token: ${{ inputs.github-token }}
+        commit-message: 'chore(version): bump to version [${{ steps.output-version.outputs.version }}] (${{ inputs.release-level }}) [BOT]'
+
+    - name: Create tag
+      uses: rickstaa/action-create-tag@a1c7777fcb2fee4f19b0f283ba888afa11678b72 # v1.7.2
+      with:
+        commit_sha: ${{ steps.commit-version-bump.outputs.commit-sha }}
+        github_token: ${{ inputs.github-token }}
+        message: "chore(tag): tag version ${{ steps.output-version.outputs.version-tag }} [BOT]"
+        tag: ${{ steps.output-version.outputs.version-tag }}
+
+    - name: Upload build artifact
+      uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+      with:
+        name: upload-dir
+        path: ${{ runner.temp }}/plugin-artifacts
+
+    # Checkout the tag, to create the release
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ steps.output-version.outputs.version-tag }}
+
+    - name: Create Github release
+      uses: softprops/action-gh-release@9d7c94cfd0a1f3ed45544c887983e9fa900f0564 # v2.0.4
+      with:
+        prerelease: ${{ inputs.release-level == 'prerelease' }}
+        generate_release_notes: true
+        files: |
+          ${{ runner.temp }}/plugin-artifacts/**/*.zip?(.md5)
+        tag_name: ${{ steps.output-version.outputs.version-tag }}
+        token: ${{ inputs.github-token }}
+
+    - id: auth
```

Comment:
I think in such a setup going with OIDC is the preferred approach in the long run, similar to what has been done here: https://github.com/grafana/shared-workflows/pull/40/files#diff-b9fbde687980aa46fe36ddbd7d7ac35be4e057ab7f6e6a102e269261fbbe8a52R44

But since the credentials are "just" an input, this should be ok in this context.

@dsotirakis WDYT?

Reply:
Hm, totally agree with @zerok. @abannachGrafana maybe it's better to do this now as part of this PR. We would like to move away from SA secrets and GitHub tokens in general since we have the OIDC auth setup already in place.

Reply:
> move away from SA secrets and GitHub tokens

I'm on board. 

Are service accounts being automatically generated for each repo or is that up to the codeowners to create them in deployment tools?

Since the credentials are inputs right now that opens up to anybody being able to leverage this action. Assuming we go the OIDC route, does that restrict this action to Grafana org use only? Happy to make the change if it simplifies workflows from having to gather creds themselves, but just want to make sure I understand the implications.

Reply:
> Assuming we go the OIDC route, does that restrict this action to Grafana org use only?

yep!

Reply:
we still don't have a mechanism to auth and push to GCS though, we are working on it atm (cc @ricky-undeadcoders)
what we can do for now is to create an SA with write perms to `${{ inputs.gcs-bucket }}` until we fix the issue. I'll create a PR.

Reply:
> > Assuming we go the OIDC route, does that restrict this action to Grafana org use only?
> 
> yep!

I don't have a horse in either race. So as long as we're good with that I'm happy with the decision. After @ricky-undeadcoders PR is merged we can update this to leverage it 👍